File: hello.js
// console.log("Hello #1190");
// console.log(typeof(12));
// console.log(typeof(13.4));
// console.log(typeof(true));
// console.log(typeof("hello"));

// // (12).toString()
// console.log(typeof((12).toString()));
// // parseInt("13", 10);
// console.log(typeof(parseInt("13", 10)));
// console.log(parseInt("101", 10));

// Count in binary
// 0 => 0
// 1 => 1
// 10 => 2
// 11 => 3
// 100 => 4
// 101 => 5
// console.log(parseInt("101", 2));

const student = "dOnAld"; // cannot be re-assigned
let age = 21; // can be re-assigned

console.log(`${student} is ${age} years-old.`);
age += 1;
console.log(age);
// student = "Super Donald"; //DOESN'T WORK
// console.log(student);
console.log(`${student} is ${age} years-old next year.`);

console.log(student.toLowerCase());
console.log(student.toUpperCase());
// no capitalize in JS!!

console.log(student.length);
console.log(student[0]);

console.log(student.slice(1)); //from index 1 to the end
console.log(student.slice(1, 3)); //from index 1 to index 3 excluded
console.log(student.substring(1, 3)); //from index 1 to index 3 excluded

const students_string = "Joe,Daniel,Ryo,Andrew";
const students_array = students_string.split(",");
console.log(students_array);
console.log(students_array.join()); // JS .join add commas by default!
console.log(students_array.join(""));
console.log(students_array.join("="));

console.log(`Multi
line
string`);

// console.log("Multi
// line
// string"); // DOESN'T WORK

File: hello.rb
# puts "Hello #1190"
# puts 12.class
# puts 13.4.class
# puts true.class
# puts "hello".class

# puts 12.to_s.class
# puts "12".to_i.class

student = "dOnAld"
age = 21

puts "#{student} is #{age} years-old."
age += 1
puts "#{student} is #{age} years-old next year."

puts student.downcase
puts student.upcase
puts student.capitalize

# puts student.count #only on an array
puts student.length
puts student.size
# puts student.first #only on an array
puts student[0]
puts student[0..2]
puts student.slice(0,3)

students_string = "Joe,Daniel,Ryo,Andrew"
# split splits on the sapces
students_array = students_string.split(",")
p students_array
p students_array.join
p students_array.join("~")